src/Evolution/Systems/ScalarAdvection/FiniteDifference/ReconstructWork.tpp

```diff
@@ -0,0 +1,161 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#pragma once
+
+#include <array>
+#include <boost/functional/hash.hpp>
+#include <cstddef>
+#include <utility>
+
+#include "DataStructures/FixedHashMap.hpp"
+#include "DataStructures/Index.hpp"
+#include "DataStructures/Variables.hpp"
+#include "Domain/Structure/Direction.hpp"
+#include "Domain/Structure/DirectionMap.hpp"
+#include "Domain/Structure/Element.hpp"
+#include "Domain/Structure/ElementId.hpp"
+#include "Domain/Structure/MaxNumberOfNeighbors.hpp"
+#include "Evolution/DgSubcell/NeighborData.hpp"
+#include "Evolution/Systems/ScalarAdvection/Tags.hpp"
+#include "NumericalAlgorithms/Spectral/Mesh.hpp"
+#include "Utilities/ErrorHandling/Assert.hpp"
+#include "Utilities/Gsl.hpp"
+#include "Utilities/TMPL.hpp"
+
+namespace ScalarAdvection::fd {
+template <size_t Dim, typename TagsList, typename Reconstructor>
+void reconstruct_work(
+    const gsl::not_null<std::array<Variables<TagsList>, Dim>*>
+        vars_on_lower_face,
+    const gsl::not_null<std::array<Variables<TagsList>, Dim>*>
+        vars_on_upper_face,
+    const Reconstructor& reconstruct,
+    const Variables<tmpl::list<Tags::U>> volume_vars,
+    const Element<Dim>& element,
+    const FixedHashMap<maximum_number_of_neighbors(Dim) + 1,
+                       std::pair<Direction<Dim>, ElementId<Dim>>,
+                       evolution::dg::subcell::NeighborData,
+                       boost::hash<std::pair<Direction<Dim>, ElementId<Dim>>>>
+        neighbor_data,
+    const Mesh<Dim>& subcell_mesh, const size_t ghost_zone_size) {
+  // check if subcell mesh is isotropic
+  ASSERT(Mesh<Dim>(subcell_mesh.extents(0), subcell_mesh.basis(0),
+                   subcell_mesh.quadrature(0)) == subcell_mesh,
+         "The subcell mesh should be isotropic but got " << subcell_mesh);
+
+  // number of points at which reconstruction is performed for each dimension,
```

Comment:
I'm not sure what `at which reconstruction` means. I would say `The number of interface points, which is equal to the number of points we reconstruct to`. Similar to interpolation, you reconstruct to a point not at a point. You can reconstruct at a cell, but I would say that's an awkward phrasing of entire polynomial reconstruction for a cell